src/solver/page/useSolverSolution.ts

```diff
@@ -0,0 +1,114 @@
+import {
+  solveProduction,
+  useHighs,
+} from '@/solver/algorithm/solveProduction';
+import {
+  useFactoryInputsOutputs,
+} from '@/factories/store/factoriesSelectors';
+import {
+  useCurrentSolverId,
+  usePathSolverInstance,
+  useSolverGameId,
+} from '@/solver/store/solverSelectors';
+import { useEffect, useMemo } from 'react';
+import { useStore } from '@/core/zustand';
+import { Path, setByPath } from '@clickbar/dot-diver';
+import { SolverInstance } from '@/solver/store/Solver';
+import { useFormOnChange } from '@/core/form/useFormOnChange';
+import {
+  ISolverSolutionSuggestion,
+  proposeSolverSolutionSuggestions,
+} from '@/solver/page/suggestions/proposeSolverSolutionSuggestions';
+import { isSolutionFound } from '@/solver/algorithm/solve/isSolutionFound';
+import { loglev } from '@/core/logger/log';
+
+const logger = loglev.getLogger('solver:page');
+
+const DEFAULT_FACTORY = {
+  inputs: [],
+  outputs: [
+    {
+      resource: 'Desc_Cement_C',
+      amount: 20,
+    },
+  ],
+};
+export const useSolverSolution = (id: string, mode: 'game' | 'standalone') => {
+  const { highsRef, loading } = useHighs();
+
+  const currentSolverId = useCurrentSolverId();
+  const inputsOutputs = useFactoryInputsOutputs(id);
+  const instance = usePathSolverInstance(id);
+  // This is not the _displayed_ solver ID, but the one that is to be used if no solver ID is provided
+  const solverGameId = useSolverGameId(id);
+
+  useEffect(() => {
```

Comment:
I should say I don't really remember exactly how this worked before and it wasn't the code I was most excited about;
I'm however worried about it, had you got the chance to test shared solvers for example? If opening one still works?
I need to fiddle with it before leaving some more useful feedback, but this is one "particular" case that comes to my mind right now

Reply:
Wdym shared solvers? Tbf this part seemed a bit nebulous to me. It took me some time to realise that standalone calculators are factories attached to no game! And that the solver are only the settings like allowed recipe and such

Reply:
I mean when you share a solver (through the "Share" button), get a link, and send it to someone else

![image](https://github.com/user-attachments/assets/a5a764be-63d3-41ea-a2fd-0bea9b622f7e)

Reply:
Seems to be working fine! Although you land on the Factory tab, and not the Calculator tab. Last commit fixes it.